### ISSUE TITLE
About backward propagation issue

### ISSUE THREAD

**Issue by TianAosheng** [LiQiufu/WaveCNet#6]
Hello,  i have some questions:
1, Are the gradient calculation methods of WResNet18_A and WResNet18_C same? Are these methods all in this way of "grad_input = torch.add(torch.matmul(grad_L, matrix_L), torch.matmul(grad_H, matrix_H))" ?
2, If so, why is it not divided by 2 in the WResNet18_A?

**Comment by LiQiufu** [comment]
WResNet18_A and WResNet18_C are implemented using torch.add and torch.cat, whose gradient calculations are automatically conducted by PyTorch. I only calculate gradient for DWT operation.
